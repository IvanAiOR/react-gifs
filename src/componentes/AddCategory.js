import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategorias}) => {
    const [NuevaCategoria, SetNuevaCategoria] = useState("Ingresa una categorÃ­a")

    const handleNewCategory = (e)=>{
        SetNuevaCategoria(e.target.value)
        
    }

    const handleSubmit = (e)=>{
        
        e.preventDefault();
        if (NuevaCategoria==='') {
            
        }
        else{
            console.log(NuevaCategoria)
            setCategorias(categorias=>[NuevaCategoria, ...categorias, ]);
        }
        SetNuevaCategoria(()=>'');
        
    }

    return (
        <form onSubmit={handleSubmit}>
            <h3>Ingresa una categoria</h3>
            <input 
                type="text"
                value={NuevaCategoria}
                onChange={handleNewCategory}
            />
            <br/>
            <button type="submit">Agregar</button>
            
        </form>
    )
}
AddCategory.propTypes= {
    setCategorias: PropTypes.func.isRequired
}
