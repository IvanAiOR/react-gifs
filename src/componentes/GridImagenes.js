import React, { useEffect, useState } from 'react'
import { GridItem } from './GridItem';
import { getGifs } from '../helpers/GetImagenes';
import { useFetchGifs } from '../hooks/useFetchGifs';

import PropTypes from 'prop-types';

export  const GridImagenes = ({categoria}) => {

    
   
    const {data:Gif,loading}=useFetchGifs(categoria);
 
    
    
    return (
        <>
        
            <h3>{categoria}</h3>
            <div className="card-grid">
                {loading && <h5>Cargando...</h5>}
                {
                    Gif.map(
                        (img)=>(
                            <
                                GridItem    
                                key={img.id}
                                {...img}

                            />
                        )
                        
                    )
                }
                
                
            </div>
        </>
    );
}

GridImagenes.prototype={
    categoria: PropTypes.string.isRequired
}