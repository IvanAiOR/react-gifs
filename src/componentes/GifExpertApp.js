import React, { useState } from 'react'
import { AddCategory } from './AddCategory';
import { GridImagenes } from './GridImagenes';

export const GifExpertApp = () => {
// const categorias =[`Gatos`, `perros`, `Anime`]
const [categorias, setCategorias] = useState(['One punch'])
    


return (
        <div>
            <h2>GifExpertApp</h2>
            <hr/>

            {/* <AddCategory /> */}
            <AddCategory setCategorias={setCategorias}/>
            <ol>
                {
                    
                    categorias.map(
                        (categoria,indice)=>{
                            console.log('categoria en gridexpertapp: ',categoria)
                            return(
                                <
                                    GridImagenes 
                                    categoria={categoria}
                                    key={categoria}
                                />
                            );
                        }
                    )
                }
            </ol>

        </div>
    )
}
