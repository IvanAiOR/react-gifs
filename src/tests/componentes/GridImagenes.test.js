import React from 'react';
import {shallow} from 'enzyme';
import { GridItem } from '../../componentes/GridItem';
import Adapter from 'enzyme-adapter-react-16';

describe('pruebas en GridImagenes', () => {
    
    const url = 'algo.jpg';
    const titulo = 'Titulo';
    
    const wrapper= shallow(<GridItem title={titulo} url={url}/>);
    
    test('Debe mostrar el componente correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe tener un parrao con el titulo', () => {
        const parrafo = wrapper.find('p');
        expect(parrafo.text().trim()).toBe(titulo);
    })
    
    test('debe tener una imagen con la url', () => {
        const img= wrapper.find('img');
        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(titulo);
    })
    
    test('el div debe contener la clase card', () => {
        const div= wrapper.find('div');
        expect(div.prop('className').includes('card')).toBe(true);
    })
    
})
