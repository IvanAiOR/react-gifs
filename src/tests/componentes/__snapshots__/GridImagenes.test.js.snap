// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GridImagenes Debe mostrar el componente correctamente  1`] = `
<div
  className="card"
>
  <img
    alt="Titulo"
    src="algo.jpg"
  />
  <p>
    Titulo
  </p>
</div>
`;
